crates/wasmtime/src/runtime/store.rs

```diff
@@ -1519,10 +1519,7 @@ impl StoreOpaque {
 
         #[cfg(feature = "gc")]
         fn allocate_gc_store(engine: &Engine) -> Result<GcStore> {
-            let (index, heap) = if engine
-                .features()
-                .contains(wasmparser::WasmFeatures::REFERENCE_TYPES)
-            {
+            let (index, heap) = if engine.features().reference_types() {
```

Comment:
While you're here mind handling https://github.com/bytecodealliance/wasmtime/pull/9162 here as well and renaming `reference_types` in this PR to `gc_types` instead?